electron/gog/library.ts

```diff
@@ -591,3 +608,89 @@ export class GOGLibrary {
     }
   }
 }
+
+/**
+ * Runs GOGDL with the given command
+ * @param commandParts The command to run, e. g. 'update', 'install'...
+ */
+export async function runGogdlCommand(
+  commandParts: Array<string>,
+  logFile?: string,
+  env = process.env
+): Promise<ExecResult> {
+  commandParts = commandParts.filter((n) => n)
+  const { bin, dir } = getGOGdlBin()
+  const safeCommand = getGogdlCommand(commandParts)
+
+  logDebug(['Running GOGDL command:', safeCommand], LogPrefix.Gog)
+  if (logFile) {
+    logDebug(['Logging to file', `"${logFile}"`], LogPrefix.Gog)
+  }
+
+  if (existsSync(logFile)) {
+    writeFileSync(logFile, '')
+  }
+
+  return new Promise((res, rej) => {
+    const child = spawn(bin, commandParts, { cwd: dir, env: env })
+
+    const stdout = new Array<string>()
+    const stderr = new Array<string>()
+
+    if (logFile) {
+      child.stdout.on('data', (data: Buffer) => {
+        appendFileSync(logFile, data.toString())
```

Comment:
Kudos for all the changes, but maybe you shouldn't use sync method. Also, new file handle is created for each chunk. This could lead to blocking of app.

I suggest opening a file once and appending to it.
It would be also nice to use node's readline module. This would enable further changes to fix the atrocity that is the `requestGameProgress` handler: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/main/electron/main.ts#L1100

Reply:
I agree that this definitely needs improvement, but I'm afraid that I don't know enough about the language yet to make these changes + they don't *really* fit into this PR. I'd say we merge this for now & then we can look into improving it.

Reply:
Sure, it can be added later.